### ISSUE TITLE
Items with quality are unusable in fabrication.

### ISSUE THREAD

**Issue by Ek-Videogames** [Regalis11/Barotrauma#7191]
This issue first appeared in modding, however upon testing this appears to happen in vanilla crafting recipes as well.
### **Description:**
Items which include a 'Quality' do not seem to be treated as valid ingredients in a crafting recipe. The material slot will glow indicating that the item is recognized, however the recipe will not allow you to start fabricating.
![image](https://user-images.githubusercontent.com/55356135/140597044-64f3ce4e-cf01-45bc-bb8f-0ead9337b390.png)

- [x] A certified _"Kraut Moment"_

**Comment by Regalis11** [comment]
This doesn't seem to be caused by the quality. The "recycle" recipes require a used fuel rod, and yours seems to be in full condition. We definitely need to communicate this more clearly though (or maybe remove the condition requirement from the recipe): there's no way to know that and it does seem like a bug.

**Comment by Ek-Videogames** [comment]
According to XML, it accepts "any" condition 0 or greater ( which does include fully intact fuel rods )
`     <Fabricate suitablefabricators="fabricator" displayname="recycleitem" requiredtime="10">
      <RequiredSkill identifier="electrical" level="15" />
      <RequiredItem identifier="uranium" />
      <RequiredItem identifier="fuelrod" mincondition="0.0" usecondition="false"/>
    </Fabricate> `
The issue of being able to 'recycle' full fuel rods is its own issue that deserves attention too. Unless I am misremembering, there is a "maxcondition" property that could be used.

Here is an image of the same recipe this time using a normal fuel rod which spawned with no 'quality level. It is at about the same condition level, however this time it is allowed to craft in fabricator. This is the "Expected Behavior" as currently exists in the XML.
![image](https://user-images.githubusercontent.com/55356135/140615892-e203b93a-08a8-4403-bc4d-a1a4d5d185ac.png)

**Comment by Regalis11** [comment]
Ah, you're right, thank you for the extra info! It seems that the problem was that the recipe's maximum condition defaults to 100%, and high-quality items with an increased max condition were considered to be above that. I tested this with empty rods, which worked as expected, so I just assumed it was caused by the recipe requiring consumed fuel rods (which it doesn't).

Fixed in https://github.com/Regalis11/Barotrauma-development/commit/8000b11ec1d8b4b68f7c26f0ddf80a628173bb9b
